docs/DeveloperGuide.md

```diff
@@ -508,27 +508,57 @@ After which, a new `AssignCommand` object will be created, and is subsequently e
+    
+11. A new `AssignCommand` will be created (using the `StudentId` or `ModuleCode` object and `Task` object created) and returned to the `LogicManager`.
+    
 12. The `LogicManager` will then call `AssignCommand#execute(Model model)`.
-13. `model#assignTaskToPerson(StudentId studentId, Task task)` method will be invoked, which will in turn invoke `model#assignTaskToPerson(StudentId studentId, Task task)` method
-    and `AddressBook#assignTaskToPerson(StudentId studentId, Task task)` method.
-14. `UniquePersonList#assignTaskToPerson(StudentId studentId, Task task)` method is called which will iterate through each `Person` object and check for matching `studentId`.
-    If no student with a matching `studentId` is found, then `PersonNotFoundException()` exception will be thrown.
-15. If a `Student` object with matching `studentId` is found the method uses `Person#isTaskAlreadyPresent(Task task)` method to check if the `task` is assigned.
-    If it is already assigned a `DuplicateTaskException()` exception is thrown.
+    
+13. If `StudentId` is used, `AssignCommand#assignTaskToPerson(StudentId studentId, Task task)` method will be invoked, 
+    which will in turn invoke `model#assignTaskToPerson(StudentId studentId, Task task)` method and 
+    `AddressBook#assignTaskToPerson(StudentId studentId, Task task)` method. If `ModuleCode` is used, 
+    `AssignCommand#assignTaskToAllInModule(ModuleCode moduleCode, Task task)` method will be invoked, which will in 
+    turn invoke `model#assignTaskToAllInModule(ModuleCode moduleCode, Task task)` method and 
+    `AddressBook#assignTaskToAllInModule(ModuleCode moduleCode, Task task)` method.
+    
+14. If `StudentId` is used, `UniquePersonList#assignTaskToPerson(StudentId studentId, Task task)` method is called.
+    If `ModuleCode` is used, `UniquePersonList#assignTaskToAllInModule(ModuleCode moduleCoded, Task task)` is called.
+    This will iterate through each `Person` object and check for matching `studentId` or `moduleCode` . 
+
+   <div markdown="span" class="alert alert-info">:information_source:
+   <b>Note:</b> 
+   If no student(s) with a matching `studentId` or `moduleCode`is found, then `PersonNotFoundException()` or `ModuleCodeNotFoundException()` will be thrown.
+   </div>
+
+
+15. If a `Student` object with matching `studentId` or `moduleCode` is found the method uses `Person#isTaskAlreadyPresent(Task task)` 
+    method to check if the `task` is assigned.
     If no similar `task` is found, the following step will take place.
+
+   <div markdown="span" class="alert alert-info">:information_source:
+   <b>Note:</b>
+   If all the student(s) has already been assigned that task, then `DuplicateTaskException()` will be thrown. 
+   If some, not all students in the `moduleCode` has already been assigned that task, then `PartialDuplicateTaskException()` will be thrown.
+   </div>
+
```

Comment:
I couldn't comment on the diagram itself, but regarding AssignCommandSequenceDiagram-1.png:

In the alt box, the condition text is overlapping with the activation bar, which might come across as a DG bug. I've faced a similar issue myself but it doesn't seem like there's a way to shift the text. A possible solution is to perhaps use another ref diagram? Just something for your consideration, though I'm sure there's probably a better solution out there.